awx/main/models/events.py

```diff
@@ -535,16 +536,19 @@ def _update_host_summary_from_stats(self, hostnames):
             # in single queries
             host_mapping = dict((summary['host_id'], summary['id']) for summary in JobHostSummary.objects.filter(job_id=job.id).values('id', 'host_id'))
             updated_hosts = set()
+            updated_hosts_list = []
             for h in all_hosts:
                 # if the hostname *shows up* in the playbook_on_stats event
                 if h.name in hostnames:
                     h.last_job_id = job.id
                     updated_hosts.add(h)
+                    updated_hosts.append(h.name)
```

Comment:
I'm not sure if this is where we want to get the hostname from. It's a good start. This will capture any hosts that are in the AWX inventory. However, there are potentially another set of hosts, that are never in the AWX inventory. There could be hosts that are added to Ansible's known hosts and automation ran against those. This is a common use case of Ansible. For example, use the ec2 modules to spin up a NEW vm. This host may never exist in the AWX inventory, but Ansible job events will have a record or running against this transient cloud host.

Reply:
In analytics, I believe we pull hostnames for automation either directly from playbook_on_stats, or from the runner_on_* task events. Ideallly, we would match that here. (cc @Ladas )

Reply:
Right now analytics is taking the host_names from `runner_on_* task` events and I am now doing the processing of `playbook_on_stats` event. 

@wenottingham  Also we're using the job's created timestamp for timestamp of first/last automation, so it might be good to align on this too?

-----

@wenottingham @chrismeyersfsu Is there a case possible when hostname would appear in `runner_on_* task` but not in `playbook_on_stats` ? (maybe when job ends in `error` state?)

Reply:
Yeah @Ladas from my observations that would be one case. Or maybe even a `failed` case if the Tower system crashes in the middle of a run, the reaper reaps the Job, and the `playbook_on_stats` event never exists for that job.

Maybe another case would be through some fancy use of Ansible `delegate_to` ?